src/borg/cache.py

```diff
@@ -553,13 +554,13 @@ def chunk_incref(self, id, stats):
         stats.update(size, csize, False)
         return ChunkListEntry(id, size, csize)
 
-    def chunk_decref(self, id, stats):
+    def chunk_decref(self, id, stats, wait=True):
         if not self.txn_active:
             self.begin_txn()
         count, size, csize = self.chunks.decref(id)
         if count == 0:
             del self.chunks[id]
-            self.repository.delete(id, wait=False)
+            self.repository.delete(id, wait=wait)
```

Comment:
makes some places in the code simpler and errors there clearer.

only if we do mass decrefs, we need async == at one place, in borg delete archive.